citadel/indico_citadel/models/search_id_map.py

```diff
@@ -0,0 +1,229 @@
+    )
+
+    #: ID of the search entry event
+    event_id = db.Column(
+        db.Integer,
+        db.ForeignKey('events.events.id'),
+        index=True,
+        nullable=True
+    )
+
+    #: ID of the search entry contribution
+    contrib_id = db.Column(
+        db.Integer,
+        db.ForeignKey('events.contributions.id'),
+        index=True,
+        nullable=True
+    )
+
+    #: ID of the search entry subcontribution
+    subcontrib_id = db.Column(
+        db.Integer,
+        db.ForeignKey('events.subcontributions.id'),
+        index=True,
+        nullable=True
+    )
+
+    #: ID of the search entry attachment
+    attachment_id = db.Column(
+        db.Integer,
+        db.ForeignKey('attachments.attachments.id'),
+        index=True,
+        nullable=True
+    )
+
+    #: ID of the search entry note
+    note_id = db.Column(
+        db.Integer,
+        db.ForeignKey('events.notes.id'),
+        index=True,
+        nullable=True
+    )
+
+    #: ID of the search entry file
+    attachment_file_id = db.Column(
+        db.Integer,
+        db.ForeignKey('attachments.files.id'),
+        index=True,
+        nullable=True,
+    )
+
+    event = db.relationship(
+        'Event',
+        lazy=True,
+        backref=db.backref(
+            'citadel_es_entries',
+            cascade='all, delete-orphan',
+            lazy=True
+        )
+    )
+
+    contribution = db.relationship(
+        'Contribution',
+        lazy=False,
+        backref=db.backref(
+            'citadel_es_entries',
+            cascade='all, delete-orphan',
+            lazy='dynamic'
+        )
+    )
+
+    subcontribution = db.relationship(
+        'SubContribution',
+        lazy=False,
+        backref=db.backref(
+            'citadel_es_entries',
+            cascade='all, delete-orphan',
+            lazy='dynamic'
+        )
+    )
+
+    attachment = db.relationship(
+        'Attachment',
+        lazy=False,
+        backref=db.backref(
+            'citadel_es_entries',
+            cascade='all, delete-orphan',
+            lazy='dynamic'
+        )
+    )
+
+    note = db.relationship(
+        'EventNote',
+        lazy=False,
+        backref=db.backref(
+            'citadel_es_entries',
+            cascade='all, delete-orphan',
+            lazy='dynamic'
+        )
+    )
+
+    attachment_file = db.relationship(
+        'AttachmentFile',
+        lazy=False,
+        backref=db.backref(
+            'citadel_es_entries',
+            cascade='all, delete-orphan',
+            lazy='dynamic'
+        )
+    )
+
+    @classmethod
+    def get_search_id(cls, oid, obj_type):
+        """Get the search_id for a given object type and id.
+
+        :param oid: The id of the object
+        :param obj_type: the EntryType of the object
+        """
+
+        obj = cls.query.filter_by(entry_type=obj_type)
+        attr = _column_for_types.get(obj_type)
+        if attr:
+            obj.filter_by(getattr(cls, attr) == oid)
```

Comment:
Not fan of the assignment here either

Reply:
Actually I just realized that this code is completely broken. `filter_by()` takes keyword ags, so it would have to be `.filter_by(**{attr: oid})` - but I think using `.filter()` with getattr or a mapping dict is less ugly...

Reply:
Fixed that in my latest - a `filter` and `getattr`.